spec/compiler/semantic/automatic_cast_spec.cr

```diff
@@ -573,9 +573,92 @@ describe "Semantic: automatic cast" do
       end
 
       Baz.new.as(Foo).foo(1)
-      )) { int64 }
+    )) { int64 }
   end
 
+  {% if flag?(:number_autocast) %}
```

Comment:
Why do we need this guard? The example gets evaluated with `number_autocast` flag, regardless of the spec compiler's flag.